recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,218 @@
-
-        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "8":
-            raise ConanInvalidConfiguration("Need Clang >= 8")
-
-        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "15.8":
-            raise ConanInvalidConfiguration("Need Visual Studio >= 2017 15.8 (MSVC 19.15)")
-
-        # Actually, we want *at least* 15.8 (MSVC 19.15), but we cannot check this for now with Conan.
-        if self.settings.compiler == "msvc" and Version(self.settings.compiler.version) < "19.15":
-            raise ConanInvalidConfiguration("Need msvc >= 19.15")
+            check_min_cppstd(self, self._minimum_cpp_standard)
+
+        def _lazy_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
+        check_min_vs(self, "192")
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if not minimum_version:
+                self.output.warn(f"{self.name} {self.version} requires C++20. Your compiler is unknown. Assuming it supports C++20.")
+            elif _lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
+                raise ConanInvalidConfiguration(f"{self.name} {self.version} requires C++20, which your compiler does not support.")
+            if self.info.settings.compiler == "clang" and str(self.info.settings.compiler.version) in ("13", "14"):
+                raise ConanInvalidConfiguration(f"{self.ref} currently does not work with Clang {self.info.settings.compiler.version} on CCI, it enters in an infinite build loop (smells like a compiler bug). Contributions are welcomed!")
 
         self._validate_task_system()
+        self._validate_thread_system()
         self._validate_boost_components()
 
     def configure(self):
         if self.options.task_system == "auto":
             self.options.task_system = self._default_task_system()
-        self.output.info("Stlab Task System: {}.".format(self.options.task_system))
+
+        if self.options.thread_system == "auto":
+            self.options.thread_system = self._default_thread_system()
+
+        self.output.info("STLab With Boost: {}.".format(self.options.with_boost))
+        self.output.info("STLab Future Coroutines: {}.".format(self.options.future_coroutines))
+        self.output.info("STLab No Standard Coroutines: {}.".format(self.options.no_std_coroutines))
+        self.output.info("STLab Task System: {}.".format(self.options.task_system))
+        self.output.info("STLab Thread System: {}.".format(self.options.thread_system))
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
```

Comment:
This seems out of place 🤔  why is it required?

Reply:
To prevent this issue:

https://c3i.jfrog.io/c3i/misc/logs/pr/13048/66-configs/windows-visual_studio/stlab/1.7.1//58aaecfc647c0c77f309fa0b9ce5e5cf8274f18e-build.txt

Reply:
It prevents this issue:

https://c3i.jfrog.io/c3i/misc/logs/pr/13048/68-configs/windows-visual_studio/stlab/1.7.1//02833e547ecc88a0495cb145970002b7a2796825-build.txt